lib/xcode.js

```diff
@@ -499,28 +491,55 @@ exports.detect = function detect(options, callback) {
 		], function () {
 			if (Object.keys(results.xcode).length) {
 				var validXcodes = 0,
+					xcodeIds = Object.keys(results.xcode),
 					sdkCounter = 0,
-					simCounter = 0;
+					simCounter = 0,
+					eulaNotAccepted = [];
 
-				Object.keys(results.xcode).forEach(function (xcodeId) {
-					results.xcode[xcodeId].sdks.forEach(function (iosVersion) {
-						if (results.xcode[xcodeId].selected || !results.iosSDKtoXcode[iosVersion]) {
+				xcodeIds.forEach(function (xcodeId) {
+					const xc = results.xcode[xcodeId];
+					xc.sdks.forEach(function (iosVersion) {
+						if (xc.selected || !results.iosSDKtoXcode[iosVersion]) {
 							results.iosSDKtoXcode[iosVersion] = xcodeId;
 						}
 					});
 
-					if (results.xcode[xcodeId].supported) {
+					if (xc.supported) {
 						// we're counting maybe's as valid
 						validXcodes++;
 					}
-					if (results.xcode[xcodeId].sdks) {
-						sdkCounter += results.xcode[xcodeId].sdks.length;
+					if (xc.sdks) {
+						sdkCounter += xc.sdks.length;
+					}
+					if (xc.sims) {
+						simCounter += xc.sims.length;
 					}
-					if (results.xcode[xcodeId].sims) {
-						simCounter += results.xcode[xcodeId].sims.length;
+					if (!xc.eulaAccepted) {
+						eulaNotAccepted.push(xc);
 					}
 				});
 
+				if (eulaNotAccepted.length) {
+					var message;
+
+					if (xcodeIds.length === 1) {
+						message = __('Xcode EULA has not been accepted.') + '\n' +
+							__('Launch Xcode and accept the license.');
+					} else {
+						message = __('Multiple Xcode versions have not had their EULA accepted:') + '\n' +
+							eulaNotAccepted.map(function (xc) {
+								return '  ' + xc.version + ' (' + xc.xcodeapp + ')';
+							}).join('\n') + '\n' +
+							__('Launch each Xcode and accept the license.');
+					}
+
+					results.issues.push({
+						id: 'IOS_XCODE_EULA_NOT_ACCEPTED',
+						type: 'warning',
```

Comment:
So, I see what you're doing here in that you're looking to see if there's multiple Codes with EULA unaccepted, and then only marking them as warnings so the build can proceed.
That would allow users to continue if say an Xcode they weren't using was unaccepted. But I assume we put this in as an error initially for a good reason. Is there no way to tell during validation what actual Xcode is going to be used? Isn't it an error if the one that they'd be using isa unaccepted? Or is that ambiguous? Maybe more than one Xcode version would potentially be used based on the target iOS?

Reply:
ioslib is just a library. It should not be the one determining what is and is not an "issue". That's why ioslib v2 has no "issues".

"issues" has always been for "ti info", not for the iOS build. However, the iOS build got a little assert happy and we full stop the build if any Xcode had an unaccepted EULA, regardless if it was the version being used to build or not.

So, this PR lets ioslib do what it's intended to do: detect and report.

The iOS build has been updated to assert the EULA has been accepted for the Xcode version being used to build: https://github.com/appcelerator/titanium_mobile/pull/10249.